locust/main.py

```diff
@@ -262,13 +262,124 @@ def parse_options():
         default=False,
         help="show program's version number and exit"
     )
+    return parser
+
+
+def create_options(
+        locustfile='locustfile',
+        host=None,
+        locust_classes=[], 
+        port=8089,
+        web_host='',
+        no_reset_stats=None,
+        reset_stats=False,
+        no_web=False,
+        run_time=None,
+        num_clients=1,
+        hatch_rate=1,
+        master=False,
+        expect_slaves=1,
+        master_bind_host='*',
+        master_bind_port=5557,
+        slave=False,
+        master_host='127.0.0.1',
+        master_port=5557,
+        csvfilebase=None,
+        print_stats=False,
+        only_summary=False,
+        show_task_ratio=False,
+        show_task_ratio_json=False,
+        logfile=None,
+        loglevel='INFO',
+        show_version=False,
+        list_commands=False):
+    """Create options objects for passing to `run_locust` when running Locust programmatically.
+    
+    Keyword Arguments:
```

Comment:
I like how all of the options are listed here, but could the most important ones be isolated out and set as kwargs in run_locust call with appropriate defaults. Other options could be grouped by their functionality like no-web options, distributed options, etc..

Reply:
Good comment; can probably apply some functional programming concepts here to reduce the complexity.

Reply:
Ok, will group these up with comments explaining the grouping.